### PR TITLE
graphql: Use a `filter` argument for more flexible filtering.

### PR DESCRIPTION
**Pull request by vpetrovykh**
Rather than taking arguments corresponding to the individual fields, use
a single `filter` argument to specify the filtering conditions. This is
more generic and easier to add more functionality to in the future.

Notice that this is based on top of the "gql-types" branch, so you may want to review that one first..

### CODE REVIEW

File: edb/lang/graphql/translator.py [hunk 1]
```diff
@@ -487,26 +482,51 @@ def get_path_prefix(self, end_trim=None):
...
+
+        if op:
+            value = self.visit(node.value)
+
+            if name_parts == 'contains':
+                value.value = f'%{value.value}%'
+            elif name_parts == 'startswith':
+                value.value = f'{value.value}%'
+            elif name_parts == 'endswith':
+                value.value = f'%{value.value}'
```
Comment (1st1) [inline edb/lang/graphql/translator.py]: These seem to be a special case of the "LIKE" operator. So I'd replace all of these with more generic "like" and "ilike" GraphQL filters.

File: edb/lang/graphql/types.py [hunk 2]
```diff
@@ -89,8 +106,13 @@ def __init__(self, edb_schema):
...
             query=query,
-            types=[objt for name, objt in self._gql_objtypes.items()
-                   if name != 'Query'],
+            types=[
+                objt for name, objt in
+                sorted(itertools.chain(self._gql_objtypes.items(),
+                                       self._gql_inobjtypes.items()),
+                       key=lambda x: x[0])
+                if name != 'Query'
+            ],
```
Comment (1st1) [inline edb/lang/graphql/types.py]: This is a bit hard to read, I'd suggest to extract `sorted(...)` into a separate assignment statement `types = sorted(...)`.

Comment (1st1) [reply edb/lang/graphql/types.py]: From the performance point of view it makes more sense to first filter types and sort them later.

Comment (vpetrovykh) [reply edb/lang/graphql/types.py]: OK, I've refactored the code a little bit to make it less obscure.

File: edb/lang/graphql/translator.py (hunk 1 above)
Comment (vpetrovykh) [reply edb/lang/graphql/translator.py]: I've removed the "contains", "startswith" and "endswith" and replaced them with "like" and "ilike".

**Comment by vpetrovykh** [comment]
I'm closing this one because I need to rebase it on "master" instead of the intermediate "gql-types".

**Pull request closed without merging**